LeaderElectionTest is flaky

```
[ERROR] Tests run: 6, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 13.808 s <<< FAILURE! - in io.kubernetes.client.extended.leaderelection.LeaderElectionTest
[ERROR] testLeaderElectionReportLeaderOnStart(io.kubernetes.client.extended.leaderelection.LeaderElectionTest)  Time elapsed: 2.846 s  <<< ERROR!
org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 

 cannot be returned by identity()
identity() should return String
***
If you're unsure why you're getting above error read on.
Due to the nature of the syntax above problem might occur because:
1. This exception *might* occur in wrongly written multi-threaded tests.
   Please refer to Mockito FAQ on limitations of concurrency testing.
2. A spy is stubbed using when(spy.foo()).then() syntax. It is safer to stub spies - 
   - with doReturn|Throw() family of methods. More in javadocs for Mockito.spy() method.

	at io.kubernetes.client.extended.leaderelection.LeaderElectionTest.testLeaderElectionReportLeaderOnStart(LeaderElectionTest.java:328)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
```

--- a/extended/src/test/java/io/kubernetes/client/extended/leaderelection/LeaderElectionTest.java
+++ b/extended/src/test/java/io/kubernetes/client/extended/leaderelection/LeaderElectionTest.java
@@ -24,6 +24,7 @@ import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
+import java.util.concurrent.Semaphore;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.Consumer;
@@ -291,6 +292,7 @@ public class LeaderElectionTest {
         () -> {
           leaderElector.run(() -> {}, () -> {});
         });
+    // TODO: Remove this sleep
     Thread.sleep(Duration.ofSeconds(2).toMillis());
     assertEquals(expectedException, actualException.get().getCause());
   }
@@ -308,23 +310,7 @@ public class LeaderElectionTest {
                 setLeaderTransitions(1);
                 setLeaseDurationSeconds(60);
               }
-            });
-    List<String> notifications = new ArrayList<>();
-    LeaderElectionConfig leaderElectionConfig = new LeaderElectionConfig();
-    leaderElectionConfig.setLock(lock);
-    leaderElectionConfig.setLeaseDuration(Duration.ofMillis(1000));
-    leaderElectionConfig.setRetryPeriod(Duration.ofMillis(200));
-    leaderElectionConfig.setRenewDeadline(Duration.ofMillis(700));
-    LeaderElector leaderElector = new LeaderElector(leaderElectionConfig);
-    ExecutorService leaderElectionWorker = Executors.newFixedThreadPool(1);
-    leaderElectionWorker.submit(
-        () -> {
-          leaderElector.run(() -> {}, () -> {}, (id) -> notifications.add(id));
-        });
-
-    Thread.sleep(Duration.ofSeconds(2).toMillis());
-
-    when(lock.get())
+            })
         .thenReturn(
             new LeaderElectionRecord() {
               {
@@ -335,7 +321,30 @@ public class LeaderElectionTest {
                 setLeaseDurationSeconds(60);
               }
             });
-    Thread.sleep(Duration.ofSeconds(2).toMillis());
+
+    List<String> notifications = new ArrayList<>();
+    LeaderElectionConfig leaderElectionConfig = new LeaderElectionConfig();
+    leaderElectionConfig.setLock(lock);
+    leaderElectionConfig.setLeaseDuration(Duration.ofMillis(1000));
+    leaderElectionConfig.setRetryPeriod(Duration.ofMillis(200));
+    leaderElectionConfig.setRenewDeadline(Duration.ofMillis(700));
+    LeaderElector leaderElector = new LeaderElector(leaderElectionConfig);
+    ExecutorService leaderElectionWorker = Executors.newFixedThreadPool(1);
+    final Semaphore s = new Semaphore(2);
+    s.acquire(2);
+    leaderElectionWorker.submit(
+        () -> {
+          leaderElector.run(
+              () -> {},
+              () -> {},
+              (id) -> {
+                notifications.add(id);
+                s.release();
+              });
+        });
+
+    // wait for two notifications to occur.
+    s.acquire(2);
 
     assertEquals(2, notifications.size());
     assertEquals("foo2", notifications.get(0));
@@ -365,12 +374,20 @@ public class LeaderElectionTest {
     leaderElectionConfig.setRenewDeadline(Duration.ofMillis(700));
     LeaderElector leaderElector = new LeaderElector(leaderElectionConfig);
     ExecutorService leaderElectionWorker = Executors.newFixedThreadPool(1);
+    Semaphore s = new Semaphore(1);
+    s.acquire();
     leaderElectionWorker.submit(
         () -> {
-          leaderElector.run(() -> {}, () -> {}, (id) -> notifications.add(id));
+          leaderElector.run(
+              () -> {},
+              () -> {},
+              (id) -> {
+                notifications.add(id);
+                s.release();
+              });
         });
 
-    Thread.sleep(Duration.ofSeconds(2).toMillis());
+    s.acquire();
     assertEquals(1, notifications.size());
     assertEquals("foo1", notifications.get(0));
   }

